Twitch/src/bot.py

```diff
@@ -28,14 +36,27 @@ def __init__(self, config):
 
 
 	def run(self):
+		"""
+		This function was based on Aidan's code, which is used mainly for the purpose of connecting to Twitch's Chat's servers and getting access to the chat.
+		The only parts borrowed from Aidan are the connection to the server, and the access to the messages in the chat. 
+		Everything else is my own creation.
+		"""
+
 		irc = self.irc
 		sock = self.socket
 		config = self.config
-		filechicken = open('twitchchat', 'w')  # pickle was very difficult to implement because twitch chat uses very weird unicode values, which weren't able to be picked up pickle
-		filechicken.seek(0,0)
-		count = 0 #this allows for the program to automatically close itself after a certain period of time
+		filechicken = open('twitchchat', 'w') 				 # pickle was very difficult to implement because twitch chat uses very weird unicode values, which pickle couldn't understand
```

Comment:
Amen. This is a common difficulty in text mining. There are ways to deal with unicode encoding in python for sure, but it can be a big frustration.